services/outputhost/ackmanager.go

```diff
@@ -196,21 +176,11 @@ func (ackMgr *ackManager) start() {
 	go ackMgr.manageAckLevel()
 }
 
-func (ackMgr *ackManager) getCurrentAckLevelOffset() (addr int64) {
+func (ackMgr *ackManager) getCurrentReadLevel() (addr storeHostAddress, seqNo common.SequenceNumber) {
 	ackMgr.lk.RLock()
-	if addrs, ok := ackMgr.addrs[ackMgr.ackLevel]; ok {
-		addr = int64(addrs.addr)
-	}
-	ackMgr.lk.RUnlock()
+	defer ackMgr.lk.RUnlock()
 
-	return
-}
-
-func (ackMgr *ackManager) getCurrentAckLevelSeqNo() (seqNo common.SequenceNumber) {
-	ackMgr.lk.RLock()
-	seqNo = ackMgr.levelOffset + ackMgr.ackLevel
-	ackMgr.lk.RUnlock()
-	return
+	return ackMgr.readLevelAddr, ackMgr.readLevel
```

Comment:
return ackMgr.readLevelAddr, ackMgr.readLevelStoreSeq // return the store seq here